tests/utils/datetime_test_suite.cpp

```diff
@@ -117,12 +117,12 @@ class datetime_test_suite : public test_suite
 
     void test_weekday()
     {
-        xlnt_assert_equals(xlnt::date(2000, 1, 1).weekday(), 6); // January 1st 2000 was a Saturday
-        xlnt_assert_equals(xlnt::date(2016, 7, 15).weekday(), 5); // July 15th 2016 was a Friday
-        xlnt_assert_equals(xlnt::date(2018, 10, 29).weekday(), 1); // October 29th 2018 was Monday
+        xlnt_assert_equals(xlnt::date(2000, 1, 1).weekday().get(), 6); // January 1st 2000 was a Saturday
```

Comment:
I don't like changes to unit tests, as this also means necessary refactoring in production code.

Reply:
Due to the reverted changes (see my other comment), the unit tests are no longer changed. However, I would highly prefer to change it to an `xlnt::optional` instead of using magic numbers like -1 for the next major release.